### DISCUSSION TITLE
Origin location incorrect after flashing, won't home

### DISCUSSION THREAD

**Discussion by deepwat3r** [mriscoc/Ender3V2S1#1261] [Q&A]
Ender3 V2, 4.2.7 board, CR touch, worked fine with stock firmware and there have been NO physical changes.  I wanted more control over various parameters so I flashed the latest mriscoc (Ender3V2-427-BLTUBL-MPC-20240125)

Now, with default settings it thinks the 0 position of the X axis is a random spot slightly right of center. When I do a Z home or an auto-mesh, it tries to move the head to the right past the stop and GRIND GRIND GRIND.  It does not go left to the stop switch like you'd expect.

Bed size is set to 230x230, X/Y min-max are set to default. Tried adjusting X max position but it would have to be reduced by a huge amount since the printer thinks "0" is actually at least 170mm from the left side of the board.

Any ideas?

**Comment by jorymorrison** [comment]
If you go into Prepare menu and increase your x axis position say by 5, does the head move to the right or left?

When you are home, where is Y?

**Comment by deepwat3r** [comment]
So if I increase the x axis position, it goes to the right.  If I power
cycle the printer and then auto-home, Y will always end up at 122.

X will vary depending on the starting point of the print head.  If I
disable steppers and physically move it to the normal front-left corner
position and then auto-home, X will physically be a bit more than halfway
to the right.  If I leave the print head in that spot and then auto-home
again, it will go to the right until it grinds.

**Reply by jorymorrison** [reply]
When you open the end stop diag, does x show closed even when you aren't at x home? I would think that may be causing it to always move right.

**Reply by deepwat3r** [reply]
Yes, x_min shows triggered at all times.  I just don't understand why, because on standard firmware the printer, including CR touch, works fine.  I moved to mriscoc so I could print a 230x230 object and it's obviously a superior firmware, but not if I can't manage to figure this out.

**Reply by jorymorrison** [reply]
Take a pic of your board wiring please. Ignore last reply, I wasn't thinking. Depending on how it's installed, may need to custom compile or rewire, I'm thinking.

**Comment by deepwat3r** [comment]
Yes actually, x_min shows triggered constantly.  Definitely not a physical problem with the switch though, as I can flash back to stock CR-touch firmware and everything works great.  Is it possible that for mriscoc firmware, CR touch needs to be installed differently from a cabling perspective?

![image](https://github.com/mriscoc/Ender3V2S1/assets/29443582/df143536-a6a7-4f12-92d8-b168d3610030)
